guides/10_Video_Processor.md

```diff
@@ -0,0 +1,115 @@
+|Safari on MacOS                                                        |13.0                    
+|iOS Safari                                                             |Not supported             
+
+## Video Processing APIs and Usage
+
+### VideoTransformDevice
+`VideoTransformDevice` allows `VideoFrameProcessor`s to be applied to a `Device`.
+`DefaultVideoTransformDevice` is the provided implementation of `VideoTransformDevice`. It needs four parameters: (1) `Logger` (2) `Device` (3) `Array<VideoFrameProcessor>` (4) `DefaultBrowserBehavior`.
+
+```typescript
+class DefaultVideoTransformDevice {
+  constructor(
+    logger: Logger,
+    device: Device,
+    processors: VideoFrameProcessor[],
+    browserBehavior: DefaultBrowserBehavior
+  )
+}
+```
+
+#### Construction and Starting Video Processing
+
+The construction of the `DefaultVideoTransformDevice` will not start the camera or start processing. The method `meetingSession.audioVideo.chooseVideoInputDevice` is needed to be called. Device controller will use the inner `Device` to acquire the source `MediaStream` and start the processing pipeline at the same frame rate as the source `MediaStream`.
+The parameters to `chooseVideoInputQuality` will be respected when device controller acquires the source `MediaStream`. 
+After video input is chosen, `meetingSession.audioVideo.startLocalVideoTile` can be called to start streaming video.
+
+
+#### Switching Device on VideoTransformDevice
+
+To switch the inner `Device` on `DefaultVideoTransformDevice`, call `DefaultVideoTransformDevice.chooseNewInnerDevice` with a new `Device`.
+`DefaultVideoTransformDevice.chooseNewInnerDevice` returns a new `DefaultVideoTransformDevice` but preserves the state of `VideoFrameProcessor`s. Then call `meetingSession.audioVideo.chooseVideoInputDevice` with the new transform device. 
+
+
+#### Stopping VideoTransformDevice
+
+To stop video processing for the chosen `DefaultVideoTransformDevice`, call `meetingSession.audioVideo.chooseVideoInputDevice` with different `DefaultVideoTransformDevice` or `null` to stop using previous `DefaultVideoTransformDevice`. Or `meetingSession.audioVideo.stopLocalVideoTile` can be used to stop the streaming.
+
+After the call, the inner `Device` will be released by device controller unless the inner `Device` is a `MediaStream` provided by users and it is the responsibility of users to handle the lifecycle. 
+
+After `DefaultVideoTransformDevice` is no longer used by device controller, call `DefaultVideoTransformDevice.stop` to release the `VideoProcessor`s and underlying pipeline. After `stop` is called, users must discard the `DefaultVideoTransformDevice`.`DefaultVideoTransformDevice.stop` is necessary to release all the resources.
+
+
+#### Receiving lifecycle notifications with observer
+
+To receive notifications of lifecycle events, `DefaultVideoTransformDeviceObserver` can be added via `addObserver` method. 
```

Comment:
via (the) (or remove 'method).    Think we are getting into more esoteric grammatical territory here tho :P